### ISSUE TITLE
__getitem__ on range and sheet

### ISSUE THREAD

**Issue by gdementen** [ZoomerAnalytics/xlwings#522]
For posterity, and per @fzumstein request, here is a comment I made on a commit of the migration guide to 0.9.

I have written my own API on top of xlwings, and supporting `wb[0]` (and `wb['Sheet1']`), as well as `sheet[:10, :2]` (and `sheet['A1:B10']`) instead of `sheet.range('A1:B10')` feels very natural in practice for me and my colleagues. We use this _**a lot**_. I think xlwings would be even better if this was supported natively. Obviously, this could be supported in addition to sheet.range(xxx), but in my opinion, it would be a mistake to not support it at all, or even to not use that syntax in examples to get an existing sheet out of a workbook, and even more so for getting a range of a sheet. I know that for workbook sheets, it makes it a little inconsistent between getting an existing sheet and creating a new one or getting the active one, but this is a trade-off I am more than happy to pay for. For consistency's sake, you could support both API and present `wb[0]` as a shortcut for `wb.sheets[0]`.

**Comment by fzumstein** [comment]
So trying to add a few points from our discussion on this:

* If we do `wb[0]`, then we also have to do `app[0]`, but both are not very comprehensible though, e.g.:
    ```
    while len(app) > 0:
        app[0].close()
    ```
  but this is:

    ```
    while len(app.books) > 0:
        app.books[0].close()
    ```

* `xw.books[0].sheets` is explicit, following the Zen of Python but also the original Excel model. While it's clear that you mostly want a sheet from a workbook, some might argue they only use e.g. names, so would rather have `xw.books[0][0]` default to `xw.books[0].names[0]` instead of `xw.books[0].sheets[0]`

* Using an IDE, it's not really taking longer to type due to autocomplete

* Consistency - and following Python's idea that there should only be one obvious way of doing things:
  ```
  >>> xw.books[0].sheets
  Sheets([<Sheet [Workbook1]Sheet1>, <Sheet [Workbook1]Sheet2>, <Sheet [Workbook1]Sheet3>])
  >>> xw.books[0].sheets[0]
  <Sheet [Workbook1]Sheet1>
  >>> xw.books[0].sheets.add()
  ```

* The following shortcut however seems reasonable and useful:

  ```
  >>> sht = xw.apps[0].book[0].sheets[0]
  >>> sht.range('A1')
  >>> sht['A1']  # seems a legible alternative to the line above
  ```

**Comment by gdementen** [comment]
Replying on each point.

* for app, I tend to agree that it is less readable. At least for this particular example. I think the main reason is that I don't tend to view an instance of Excel as mostly a list of workbooks with some metadata. 

* The Zen of Python also states that practicality beats purity :). Your example seems a bit far-fetched. I consider names to be "extra" data, not the core business of the object. Besides, using .sheets feels redundant, like it would if, for example, for accessing an item of a python dict we would have to use: `d.values[key]` instead of `d[key]`. Yes, that would be more explicit, but would you really want that?

* The IDE argument is a bit weak and not really true anyway because you have to type a few keystrokes before auto-completion triggers and you have the time it takes to select the correct option (even if it turns out to be the one selected by default, you still have to check that). I view auto-completion more as a way to avoid typos than to save time typing (except for the unusually long names).

* That is more concerning (and I acknowledged that right from the start). The main question I think is whether you (and people in general) consider a workbook as mainly an ordered collection of sheets plus some metadata, and that is exactly how I view those (and I think my colleagues have the same opinion). BTW, your example does not look as bad when you use sheet names instead of sheet numbers or when you store intermediate values in a variable, which I almost always do, both at the interactive prompt and in my scripts because I do several operations on the same workbook and/or sheet.

```python
>>> wb = xw.books[0]
>>> list(wb)
Sheets([<Sheet [Workbook1]Sheet1>, <Sheet [Workbook1]Sheet2>, <Sheet [Workbook1]Sheet3>])
>>> wb[0]
<Sheet [Workbook1]Sheet1>
>>> sheet = wb[0]
# does not seems too awful
>>> sheet = wb.sheets.add()
# seems better to me but would require to re-change a good chunk of your brand new API ;-)
>>> sheet = wb.add_sheet()
```

* glad that you seem to accept that one as it is the most important to me (I use it so often that the extra keystrokes would add up).

BTW, thanks for the open-minded discussion. Even if you finally decide against `sheet = wb['Sheet1']`, at least you are open to the discussion, which is very much appreciated.

**Comment by fzumstein** [comment]
Deciding over an API is never easy especially if there are two worlds involved (Excel/Python). Range shortcuts are available now: c76b3be0a11bd88b6d1ec52a35a242ba3e1886e0

However, we are leaving sheets as is for now.
